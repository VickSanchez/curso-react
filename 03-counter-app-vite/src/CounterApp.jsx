import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ( { value } ) => {

    const saludar = () => {
        console.log("Hola");
    }

    const [counter, setCounter] = useState( value );

    const handleAdd = () => {
        setCounter(counter * 2);
    }
    const handleSubstract = () => {
        setCounter( counter - 1);
    }
    const handleReset = () => {
        setCounter(value);
    }
 

    return (
        <>
        <h1>RTP App</h1>
        <h2>{ counter }</h2>

        <button onClick={ handleAdd }>Click X2</button>
        <button onClick={ handleSubstract }>Click -1</button>
        <button aria-label= "btn-reset" onClick={ handleReset }>Reset</button>



        </>
    )
};

CounterApp.propTypes ={
    value: PropTypes.number
};
