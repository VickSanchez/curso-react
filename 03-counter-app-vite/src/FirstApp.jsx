import PropTypes from "prop-types";

const newMessage = {
    message: "Hola Mundo",
    title: "Fernando",
}
const message = "Hola soy una variable"
const getMessage = () => "Vick";

export const FirstApp = ( {title, subTitle, name} ) => {

    return (
        <>
        <h1>{ title }</h1>
        <p>{ subTitle }</p>
        <p>{ name }</p>
        </>
    )
};

FirstApp.propTypes ={
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    name: "Victor",
    title: "Hola RTP",
    subTitle: 123
}