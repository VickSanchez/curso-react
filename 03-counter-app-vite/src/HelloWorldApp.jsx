export function App() {
    return <h1>Hola Mundo !!!!!</h1>;
}

export const HelloWorldApp = () => {
  return (
    <div>Hello World App</div>
  )
}

export const Msj_Bienvenida = () => <h3>Bienvenidos</h3>;