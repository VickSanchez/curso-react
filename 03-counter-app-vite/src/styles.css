
html, body {
    background-color: rgb(0, 0, 0);
    color:green;
    font-family:'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 20px;
    font-size: 1.3rem;
    font-family:'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    margin-right: 20px;
    margin-top:  20px;
    background-color: #eee;
    border: 2px solid #eee;
    border-radius: 5rem;
    cursor: pointer;
    transition: all 1s ease;
    box-shadow: inset;
}