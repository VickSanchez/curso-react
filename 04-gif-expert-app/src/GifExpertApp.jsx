import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
    
    const [ categories, setCategories ] = useState([ '' ]);
    
    const onAddCategory = ( newCategory ) => {
        if ( categories.includes(newCategory) ) return;
        setCategories([ newCategory, ...categories ]);
    }
    

    return (
        <>

            <h3> RTP </h3>

            <h1> Gif Search APP </h1>

    
            <AddCategory 
                onNewCategory={ (value) => onAddCategory(value) }
            />

            { 
                categories.map( ( category ) => (
                    <GifGrid 
                        key={ category } 
                        category={ category } />
                ))
            }




        </>
    )
}
