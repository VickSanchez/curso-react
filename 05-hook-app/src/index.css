body {
    padding: 20px;
}

button {
    margin-right: 10px;
}
h1 {
    color: green;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
}

span {
    font-family:Arial, Helvetica, sans-serif;
    font-size: 1rem;
}

a {
    margin-right: 5px;
}