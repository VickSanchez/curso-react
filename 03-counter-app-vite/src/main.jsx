import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
// import { App, HelloWorldApp, Msj_Bienvenida } from './HelloWorldApp';


ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
            {/* <App />
            <HelloWorldApp />
            <Msj_Bienvenida /> */}
            <FirstApp /> 
    </React.StrictMode>


);

