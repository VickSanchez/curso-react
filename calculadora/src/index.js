import React from 'react'
import ReactDOM from 'react-dom/client'

import { AppCalculadora } from './AppCalculadora'
import { Calcprovider } from './context/CalcProvider'
import './styles.css'



ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode> 
  <Calcprovider>
    <AppCalculadora/>
  </Calcprovider>
  </React.StrictMode>
)




