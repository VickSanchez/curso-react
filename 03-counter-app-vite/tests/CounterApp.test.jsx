import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';


describe('Pruebas en el <CounterApp />', () => {

    const initialValue = 10;
    
    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<CounterApp value={ initialValue } />);
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el valor inicial de 100 <CounterApp value={100}>', () => {
        
        render( <CounterApp value={ 100 } /> );
        expect( screen.getByText(100) ).toBeTruthy();
        // expect( screen.getByRole('heading',{ level: 2}).innerHTML ).toContain('100')

    });

    test('Debe de multiplicar con el botón X2', () => {
        
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('Click X2') )
        expect( screen.getByText('20') ).toBeTruthy();

    });

    test('Debe de decrementar con el botón -1', () => {
        
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('Click -1') );
        expect( screen.getByText('9') ).toBeTruthy();

    });

    test('Debe de funcionar el botón de Reset', () => {
        
        render( <CounterApp value={ 355 } /> );
        fireEvent.click( screen.getByText('Click X2') );
        fireEvent.click( screen.getByText('Click X2') );
        fireEvent.click( screen.getByText('Click X2') );
        // fireEvent.click( screen.getByText('Reset') );
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        expect( screen.getByText( 355 ) ).toBeTruthy();

    });




});