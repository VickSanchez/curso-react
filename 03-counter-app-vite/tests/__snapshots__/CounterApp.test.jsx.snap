// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    RTP App
  </h1>
  <h2>
    10
  </h2>
  <button>
    Click X2
  </button>
  <button>
    Click -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
